.github/workflows/reproducible-build.yml

```diff
@@ -86,6 +90,42 @@ jobs:
           fetch-depth: 0
           repository: ${{ env.GITHUB_REPOSITORY }}
 
+      - name: Setup Credentials
```

Comment:
We understand the need of a closed-source contract to depend on private dependencies (of course, such contracts would not pass the _verified and reproducible build_ check) :+1: 

Though, how would you feel about the options below?

(A) have your private dependencies brought as git submodules in your main private repository, then alter the `Cargo.toml` to point to them (of course, making your project open-source would reveal the source code of the dependencies, as well).

(B) in the Docker images for reproducible builds, we can enable `ssh-agent`, and run `ssh-add` (if any key is provided in the Docker environment) - I did not yet test this approach, though.

(C) allow `reproducible-build.yml` to be parametrized with the fully qualified-name of a Docker image (instead of `image_tag` only) - this way, you can use a custom Docker image to perform the build of a closed-source contract.